.github/workflows/ci.yaml

```diff
@@ -52,15 +53,24 @@ jobs:
       - name: Install dependencies
         run: |
           cd ./garden-backend-service
-          pip install --upgrade pip
           poetry install --no-interaction --no-ansi --with=develop
 
-      - name: Test with pytest
+      - name: Check for vulnerable libraries with safety
         run: |
           cd ./garden-backend-service
-          poetry run pytest
+          poetry run safety check
 
-      - name: Check for vulnerable libraries with safety
+      - name: Format with Ruff
         run: |
           cd ./garden-backend-service
-          poetry run safety check
+          poetry run ruff format
```

Comment:
I think we actually don't want a format step in the action, just the lint step -- what happens if e.g. I don't have pre-commit hooks installed and submit an unformatted pull request?  
  1. code is unformatted on my branch 
  2. `Format with Ruff` step runs, the code is formatted on the worktree inside this GH action (but not in any commit on my branch)
  3. `Lint with Ruff` step runs, sees only the formatted code from step 2 and doesn't complain 
  4. Build succeeds and we merge from my unformatted branch
  
Also, philosophically, I don't think we would actually want a CI pipeline to change the code at all (even superficially) once it's been reviewed.

Reply:
we definitely do want the precommit hook to run the formatter though

Reply:
That makes sense!